// Copyright (c) 2022 Boston Dynamics, Inc.  All rights reserved.
//
// Downloading, reproducing, distributing or otherwise using the SDK Software
// is subject to the terms and conditions of the Boston Dynamics Software
// Development Kit License (20191101-BDSDK-SL).

syntax = "proto3";

package bosdyn.api;

option java_outer_classname = "AuthServiceProto";

import "bosdyn/api/auth.proto";

// The AuthService provides clients the ability to convert a user/password pair into a token. The
// token can then be added to the http2 headers for future requests in order to establish the
// identity of the requester.
service AuthService {
    // Request to get the auth token for the robot.
    rpc GetAuthToken(GetAuthTokenRequest) returns (GetAuthTokenResponse) {}
}
